docs/markdown/concepts-and-architecture/components.md

```diff
@@ -0,0 +1,125 @@
+
+In practice, we use ethereum events as the medium for sending Ethereum RPC data from Ethereum to Polkadot.
+
+```
+PolkadotRPC {
+    sourceContractAddress EthereumContractAddress
+    targetPallet Int
+    payload Bytes
+}
+```
+
+Payloads for a PolkadotRPC are expected to be RLP encoded, as emitted by Ethreum and the receiving pallet is expected to handle and decode it into the Rust types it needs.
+
+### <a name="ethereum-light-client-verifier"></a> Ethereum Light Client Verifier
+The Ethereum Light Client Verifier is a pallet that is responsible for performing light client verification of Ethereum consensus and state on-chain on the parachain.
+
+It follows the longest Ethereum chain, verifies proof of work and tracks accepted and confirmed headers beyond a given number of confirmations.
+
+It is also responsible for accepting and verifying incoming ethereum events. It does so using transaction receipt proofs which prove that a particular transaction to a particular Ethereum smart contract was in fact valid, was included in the chain, and did emit some ethereum events. It accepts and processes a proof, verifies it and then returns the set of Ethereum events that were emitted by the proven transaction receipt.
+
+For more details, see [Ethereum Light Client Verifier](./ethereum-light-client-verifier)
+
+### Polkadot-to-Ethereum Direction
+
+#### Ethereum RPC
+An Ethereum RPC is a data structure that is used to represent asynchronous calls that a Polkadot pallet wants to make on a remote Ethereum smart contract. They are generated by a Polkadot pallet and then sent over the bridge to a consuming smart contract. They are asynchronous, one-way calls that do not require any response. They are received and processed by Ethereum Smart Contracts. 
+
+In practice, we use parachain message commitments, as described below, as the medium for sending Ethereum RPC data from Polkadot to Ethereum.
+
+
+```
+EthereumRPC {
+    sourcePallet Int
+    targetContractAddress EthereumContractAddress
+    payload Bytes
+}
+```
+
+Payloads for an EthereumRPC are expected to be ABI encoded contract call data which have been encoded by the sending pallet before sending. The receiving smart contract will be called directly with that ABI-encoded call.
+
+#### Snowbridge Parachain
+With our bridge running on Polkadot, we aren't just verifying and sending messages on the Polkadot relay chain. We also want to be able to send and verify messages on any parachain, and support assets and functionality that are not supported on the relay chain. Snowbridge runs its own parachain to facilitate this.
+
+For more details, see [Snowbridge Parachain](./snowbridge-parachain)
+
+#### <a name="parachain-message-commitment"></a> Parachain Message Commitment
```

Comment:
Shouldn't be necessary to add these anchors manually. The Markdown processor will generate an anchor automatically by downcasing the header text.

So can use something like this to link to this section:

```markdown
blah blah [foo](./components#parachain-message-commitment) blah blah
```

The anchor `#parachain-message-commitment` should be generated automatically by the Markdown processor by downcasing the